common/lcs/tests/serde.rs

```diff
@@ -287,6 +290,34 @@ fn sequence_not_long_enough() {
     assert_eq!(from_bytes::<Vec<u8>>(&seq), Err(Error::Eof));
 }
 
+#[test]
+fn map_not_canonical() {
+    let mut map = BTreeMap::new();
+    map.insert(4u8, ());
+    map.insert(5u8, ());
+    let seq = vec![2, 0, 0, 0, 4, 5];
+    assert_eq!(from_bytes::<BTreeMap<u8, ()>>(&seq), Ok(map));
+    // Make sure out-of-order keys are rejected.
+    let seq = vec![2, 0, 0, 0, 5, 4];
+    assert_eq!(
+        from_bytes::<BTreeMap<u8, ()>>(&seq),
+        Err(Error::NonCanonicalMap)
+    );
+}
+
+#[test]
+fn by_default_btreesets_are_serialized_as_sequences() {
```

Comment:
Hmm I guess i never thought about sets but really they probably should be serialized/deserialized as `Map<Key, ()>` in order to preserve the uniqueness guarantees.

Reply:
Yes, `Map<Key, ()>` works. It is in the unit test just above!